### PR TITLE
feat: add file-based input and output to faucet commands

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
In this PR the following functionality were implemented:
- Added file-based I/O to faucet commands
- Combined faucet script and metadata signature generation steps
- Safe-guarded faucet commands:
  - wallet will exit immediately afterwards;
  - faucet commands will only be allowed in command mode.

The screen output running the m-of-n spend scenario is shown below.

**Step 1 - FaucetGenerateSessionInfo**
```
faucet-generate-session-info --fee-per-gram 5 --commitment b8b0ad44a4ac05d7c383ac6615c0ed7b14d0f25de42210e89f3191e605865361 
--output-hash 7018f30e8d6bbdce7d79709db9854589b0b645ed25624bc7a6b1be43a2eded26 
--recipient-address f48yg4or3d7AsKhVD5124wSzjEE7A1ZbANhyCsJJdmKF8jcfZCcoWiGRxeyauriwPPTMswxnpdUfdWtQ4qkaDgwPrsk 
--verify-unspent-outputs
Initializing logging according to "node_04\\config\\log4rs_console_wallet.yml"
Minotari Console Wallet running... (Command mode started)
==============
Command Runner
==============

1. FaucetGenerateSessionInfo(FaucetGenerateSessionInfoArgs { fee_per_gram: MicroMinotari(5), commitment: 
"b8b0ad44a4ac05d7c383ac6615c0ed7b14d0f25de42210e89f3191e605865361", output_hash: 
"7018f30e8d6bbdce7d79709db9854589b0b645ed25624bc7a6b1be43a2eded26", recipient_address: 
Dual(DualAddress { network: Esmeralda, features: TariAddressFeatures(3), public_view_key: 
5a658abfa39fb6828b3c267fdfabc4b764aab155d816dfac774649a83649071e, public_spend_key: 
7cf45a38904b23ca76ecd95aee5c1742fc1bf6fc0ef8cd6ee2ee28540f9edb40 }), verify_unspent_outputs: true })


Concluded step 1 'faucet-generate-session-info'
Your session ID is:                 'MakczWU9X2RgPXdz'
Your session's output directory is: '<user>\AppData\Local\tari_faucets\MakczWU9X2RgPXdz'
Session info saved to:              '<user>\AppData\Local\tari_faucets\MakczWU9X2RgPXdz\step_1_session_info.json'
Send 'step_1_session_info.json' to parties for step 2

Minotari Console Wallet running... (Command mode completed)

Shutting down wallet... Done.
```
**Step 2 - FaucetCreatePartyDetails**
```
faucet-create-party-details --input-file "<user>\AppData\Local\tari_faucets\step_1_session_info.json" --alias alice
Initializing logging according to "node_02\\config\\log4rs_console_wallet.yml"
Minotari Console Wallet running... (Command mode started)
==============
Command Runner
==============

1. FaucetCreatePartyDetails(FaucetCreatePartyDetailsArgs { session_id: "MakczWU9X2RgPXdz", input_file: "
<user>\\AppData\\Local\\tari_faucets\\step_1_session_info.json", alias: "alice" })


Concluded step 2 'faucet-create-party-details'
Your session's output directory is <user>\AppData\Local\tari_faucets\MakczWU9X2RgPXdz'
Session info file '<user>\AppData\Local\tari_faucets\step_1_session_info.json' moved to '<user>\AppData\Local\tari_faucets\MakczWU9X2RgPXdz\step_1_session_info.json'
Send 'step_2_for_leader_from_alice.json' to leader for step 3

Minotari Console Wallet running... (Command mode completed)

Shutting down wallet... Done.
```
**Step 3 - FaucetEncumberAggregateUtxo**
```
faucet-encumber-aggregate-utxo --session-id MakczWU9X2RgPXdz --input-file-names=step_2_for_leader_from_alice.json
Initializing logging according to "node_04\\config\\log4rs_console_wallet.yml"
Minotari Console Wallet running... (Command mode started)
==============
Command Runner
==============

1. FaucetEncumberAggregateUtxo(FaucetEncumberAggregateUtxoArgs { session_id: "MakczWU9X2RgPXdz", input_file_names: ["step_2_for_leader_from_alice.json"] })


Concluded step 3 'faucet-encumber-aggregate-utxo'
Send 'step_3_for_parties.json' to parties for step 4

Minotari Console Wallet running... (Command mode completed)

Shutting down wallet... Done.
```
**Step 4 - FaucetCreateInputOutputSigs**
```
faucet-create-input-output-sigs --session-id MakczWU9X2RgPXdz
Initializing logging according to "node_02\\config\\log4rs_console_wallet.yml"
Minotari Console Wallet running... (Command mode started)
==============
Command Runner
==============

1. FaucetCreateInputOutputSigs(FaucetCreateInputOutputSigArgs { session_id: "MakczWU9X2RgPXdz" })


Concluded step 4 'faucet-create-input-output-sigs'
Send 'step_4_for_leader_from_alice.json' to leader for step 5

Minotari Console Wallet running... (Command mode completed)

Shutting down wallet... Done.
```
**Step 5 - FaucetSpendAggregateUtxo**
```
faucet-spend-aggregate-utxo --session-id MakczWU9X2RgPXdz --input-file-names=step_4_for_leader_from_alice.json
Initializing logging according to "node_04\\config\\log4rs_console_wallet.yml"
Minotari Console Wallet running... (Command mode started)
==============
Command Runner
==============

1. FaucetSpendAggregateUtxo(FaucetSpendAggregateUtxoArgs { session_id: "MakczWU9X2RgPXdz", input_file_names: ["step_4_for_leader_from_alice.json"] })


Concluded step 5 'faucet-spend-aggregate-utxo'

Minotari Console Wallet running... (Command mode completed)

Shutting down wallet... Done.
```

Motivation and Context
---
Manual inputs are cumbersome and prone to errors.

How Has This Been Tested?
---
System-level testing

What process can a PR reviewer use to test or verify this change?
---
Review code changes
Perform system-level testing

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

File: base_layer/wallet/src/output_manager_service/handle.rs [hunk 1]
```diff
@@ -165,7 +165,7 @@ impl fmt::Display for OutputManagerRequest {
                 "Encumber aggregate utxo with tx_id: {} and output: ({},{})",
                 tx_id,
                 expected_commitment.to_hex(),
-                output_hash
+                output_hash.to_hex()
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/handle.rs]:
```suggestion
                output_hash
```

HashOutput already implements to_hex for display

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 2]
```diff
@@ -1203,13 +1203,17 @@ where
...
-            FixedHash::from_hex(&output_hash).map_err(|e| OutputManagerError::ConversionError(e.to_string()))?;
         let output = self
             .fetch_unspent_outputs_from_node(vec![output_hash])
             .await?
             .pop()
-            .ok_or_else(|| OutputManagerError::ServiceError(format!("Output not found (TxId: {})", tx_id)))?;
+            .ok_or_else(|| {
+                OutputManagerError::ServiceError(format!(
+                    "Output with hash {} not found in blockchain (TxId: {})",
+                    output_hash.to_hex(),
```
Comment (SWvheerden) [inline base_layer/wallet/src/output_manager_service/service.rs]:
```suggestion
                    output_hash,
```

File: base_layer/wallet/src/transaction_service/handle.rs [hunk 3]
```diff
@@ -244,7 +247,7 @@ impl fmt::Display for TransactionServiceRequest {
                  script_input_shares = {:?},, script_signature_shares = {:?}, sender_offset_public_key_shares = {:?}, \
                  metadata_ephemeral_public_key_shares = {:?}, dh_shared_secret_shares = {:?}, recipient_address = {}",
                 fee_per_gram,
-                output_hash,
+                output_hash.to_hex(),
```
Comment (SWvheerden) [inline base_layer/wallet/src/transaction_service/handle.rs]:
```suggestion
                output_hash,
```

File: applications/minotari_console_wallet/src/cli.rs [hunk 4]
```diff
@@ -166,123 +163,49 @@ pub struct BurnMinotariArgs {
...
-    #[clap(long)]
-    pub m: u8,
+    pub commitment: String,
     #[clap(long)]
-    pub message: String,
+    pub output_hash: String,
     #[clap(long)]
-    pub maturity: u64,
+    pub recipient_address: TariAddress,
     #[clap(long)]
```
Comment (SWvheerden) [inline applications/minotari_console_wallet/src/cli.rs:175]:
```suggestion
    #[clap(long, default_value=false)]
```
So we dont have to add it every time

File: applications/minotari_console_wallet/src/automation/commands.rs [hunk 5]
```diff
@@ -818,178 +916,184 @@ pub async fn command_runner(
...
 
+                let mut script_signature = Signature::default();
                 match key_manager_service
-                    .sign_with_nonce_and_message(&args.private_key_id, &args.secret_nonce_key_id, &challenge)
+                    .sign_with_nonce_and_message(
+                        &party_info.wallet_spend_key_id,
+                        &party_info.script_nonce_key_id,
+                        &challenge,
+                    )
                     .await
```
Comment (SWvheerden) [inline applications/minotari_console_wallet/src/automation/commands.rs:991]: nit: this is fine as is, but we should be able to just exit here with `?` as there is no point in carrieing if there is a problem with the signature, same goes for below. Doing this we don't have to check if they are default below.

**Review by SWvheerden — DISMISSED** [review]

File: applications/minotari_console_wallet/src/cli.rs (hunk 4 above)
Comment (hansieodendaal) [reply applications/minotari_console_wallet/src/cli.rs:175]: It is switch, therefor already default false

File: applications/minotari_console_wallet/src/automation/commands.rs (hunk 5 above)
Comment (hansieodendaal) [reply applications/minotari_console_wallet/src/automation/commands.rs:991]: I opted for this to give comprehensice feedback to the user.

**Review by hansieodendaal — COMMENTED** [review]
- User feedback for `FaucetEncumberAggregateUtxo` in the wrong scope
- `FaucetCreatePartyDetails` can read the session ID from the supplied file, thus it is not needed as a command line argument.

**Pull request merged**